frontend/catalyst/compilation_pipelines.py

```diff
@@ -452,31 +417,21 @@ class QJIT:
         compile_options (Optional[CompileOptions]): Common compilation options
     """
 
-    def __init__(self, fn, target, keep_intermediate, compile_options=None):
+    # pylint: disable=too-many-arguments
+    def __init__(self, fn, target, keep_intermediate, passes, compile_options=None):
```

Comment:
https://github.com/PennyLaneAI/catalyst/pull/38/commits/f46294eb17de3397ebb01f42ba1b6955813381c9 I think moving arguments to `compile_options` does have its merits, but I don't like that when testing internal components, I will need to construct a CompileOptions object. Very minor point.

Reply:
I see your point, although that is the solution we settled on in order to prevent our function signatures from ballooning.
Maybe what we could do is define the `__get_item__` method and then use it interchangeably as a dictionary inside our code, and for testing one can also pass a dictionary rather than the CompileOptions object. Not sure it's that much of a difference though.

Reply:
Is https://github.com/PennyLaneAI/catalyst/commit/f46294eb17de3397ebb01f42ba1b6955813381c9 relevant to this discussion?

Reply:
It changes the docstring.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/38/commits/d03637db6c92a023177218ce2ecad73be0d69b10